src/context/AnalyticsContext.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useMemo,
+} from 'react';
+import { useBeamer } from '~hooks/useBeamer';
+
+// Define the shape of the analytics context
+interface AnalyticsEvent {
+  event: string;
+  category: string;
+  action: string;
+  label?: string;
+  value?: number;
+}
+
+interface AnalyticsContextValue {
+  trackEvent: (event: AnalyticsEvent) => void;
+  trackPageView: (path: string) => void;
+}
+
+// Create a context with a default implementation (no operation)
+export const AnalyticsContext = createContext<AnalyticsContextValue>({
+  trackEvent: () => {},
+  trackPageView: () => {},
+});
+
+// AnalyticsContextProvider component
+export const AnalyticsContextProvider: FC<PropsWithChildren> = ({
+  children,
+}) => {
+  const gtmId = process.env.GOOGLE_TAG_MANAGER_ID; // GTM ID from environment variable
+  useBeamer();
+
+  useEffect(() => {
+    // Inject GTM script only if gtmId is available
+    if (gtmId && (!window.dataLayer || window.dataLayer.length === 0)) {
```

Comment:
`gtmId` should short-circuit early, rather than having repetitive checks all over the place

Reply:
Good pickup, I used to have that in, but forgot to add it back in after losing my updates.